### DISCUSSION TITLE
About README.md

### DISCUSSION THREAD

**Discussion by SarveshDevrukhkar** [kunalkeshan/SRMIST-B.Tech-ECE-Notes-2022-24#148] [Q&A] [answered]
Hello @kunalkeshan,
Many times I have contributed to this project and every time I need to fork it again.

**Situation** is when you merge PR which was created by me, I need to sync it again.

No problem in syncing, but why that GitHub does one commit automatically, `"Merge branch 'client' of github.com:kunalkeshan/SRMIST-..."`.

Why is it needed?
Because in the end, I want to sync with your branch. Then why that commit?

The reason I am asking is that the next time I am going to contribute(opening PR), it shows `your branch is N commits ahead of upstream` and I don't want that.

**Comment by kunalkeshan** [comment]
The only way is to keep in Sync with the main repo. The reason is that myself and others contribute as well. So it would be best if you considered the newly updated contributions along with the ones you made to ensure that your changes also work.

**Reply by SarveshDevrukhkar** [reply]
Means I need to re-fork every time! Right?

**Reply by kunalkeshan** [reply]
You just need to fork it once. And sync updates before contributing to the same repository. 

eg: 

![image](https://github.com/kunalkeshan/SRMIST-B.Tech-ECE-Notes-2022-24/assets/68579547/3761a209-2967-4fad-becd-ec19b1a95894)

**Reply by SarveshDevrukhkar** [reply]
_If I sync the fork then GitHub will add an extra commit of `merging` and then again If I create the PR/contribute to your project, then that unnecessary commit of `merging` will get added/merged into your project._

Am I Right? (Actually, This is big confusion of mine)

**Comment by kunalkeshan** [comment, marked as answer]
Yes, it will add the commit from the source repo. In the next contribution, creating a branch from your `main` repo is best recommended and merging it into the `main` branch of the source repo to avoid conflicts. 

This allows you to avoid adding additional commits to the same branch, (because sometimes the repo maintainer might use squash and merge) and also make sure that you receive the same commits from the source repo for the `main` branch.

**Comment by SarveshDevrukhkar** [comment]
💬 Chat bubble
🗨️ Left speech bubble
💭 Thought bubble
🗣️ Speaking head
📢 Loudspeaker
📣 Megaphone
📝 Memo
💡 Light bulb
🗣️ People talking
👥 People
🤝 Handshake
👍 Thumbs up

I was thinking of adding Emoji before the `Feedback` section in `README.md` of `main`.
So, I searched for commonly used emojis for **Feedback**, and there I suggested the **💬 Chat bubble** emoji.

*So, Can I open PR with these changes?*

> 😅and I want to become a contributor!😁

**Reply by kunalkeshan** [reply]
Yeah Sure, feel free to do it. 
The 💬 does the trick, thanks!

**Reply by SarveshDevrukhkar** [reply]
> Yeah Sure, feel free to do it. 
> The 💬 does the trick, thanks!

Finally, Let's Goooooooooo>>>>🤩😇🎉🎊🎇🎆🏇🤾‍♂️⛷️🏄‍♀️🚄
